r/R/dplyr-funcs-datetime.R

```diff
@@ -147,5 +147,12 @@ register_bindings_datetime <- function() {
   register_binding("pm", function(x) {
     !call_binding("am", x)
   })
-
+  register_binding("tz", function(x) {
+    if (call_binding("is.character", x) ||
+        call_binding("is.Date", x)) {
+      return("UTC")
+    } else {
+      return(x$type()$timezone())
+    }
```

Comment:
What happens if you call this binding on an integer or a float?

Reply:
It abandons ship and pulls data into R, when `lubridate` returns `"UTC"`.

Reply:
We could go back to my first variant, which was also the simplest:
```r
  register_binding("tz", function(x) {
      x$type()$timezone()
  })
```

I can't figure out how to surface an error from `tz()` as it gets caught by the dplyr query and then the dplyr-equivalent expression still gets evaluated in R.

Reply:
I actually think that's ok — the arrow types are a bit more confusing than the R types, _but_ they are more accurate for what the issue is.